dfastmi/io/map_file.py

```diff
@@ -0,0 +1,347 @@
+        )
+        if len(mesh2d) != 1:
+            raise ValueError(
+                "Currently only one 2D mesh supported ... this file contains {} 2D meshes.".format(
+                    len(mesh2d)
+                )
+            )
+        
+        return mesh2d[0]
+
+
+    @staticmethod
+    def _copy_var(source_dataset: nc.Dataset, var_name: str, target_dataset: nc.Dataset) -> None:
+        """
+        Copy a single variable from one netCDF file to another.
+
+        Copy a single netCDF variable including all attributes from source file to
+        destination file. Create dimensions as necessary.
+
+        Arguments
+        ---------
+        source_dataset : netCDF4.Dataset
+            Dataset object representing the source file.
+        var_name : str
+            Name of the netCDF variable to be copied from source to destination.
+        target_dataset : netCDF4.Dataset
+            Dataset object representing the destination file.
+        """
+        # locate the variable to be copied
+        variable = source_dataset.variables[var_name]
+
+        # copy dimensions
+        for dim_name in variable.dimensions:
+            dimension = source_dataset.dimensions[dim_name]
+            if dim_name not in target_dataset.dimensions.keys():
+                target_dataset.createDimension(
+                    dim_name, (len(dimension) if not dimension.isunlimited() else None)
+                )
+
+        # copy variable
+        variable_copy = target_dataset.createVariable(var_name, variable.datatype, variable.dimensions)
+
+        # copy variable attributes all at once via dictionary
+        variable_copy.setncatts(variable.__dict__)
+        variable_copy[:] = variable[:]
+
+    def copy_ugrid(self, target_file: Path) -> None:
+        """
+        Copy UGRID mesh data from one netCDF file to another.
+
+        Copy UGRID mesh data (mesh variable, all attributes, all variables that the
+        UGRID attributes depend on) from source file to destination file.
+
+        Arguments
+        ---------
+        target_file : Path
+            Path to the target file.
+        """
+        target_file.unlink(missing_ok=True)
+        
+        with nc.Dataset(self._map_file) as source_dataset:
+            with nc.Dataset(target_file, "w", format="NETCDF4") as target_dataset:
+
+                mesh_variable = source_dataset.variables[self.mesh2d_name]
+
+                MapFile._copy_var(source_dataset, self.mesh2d_name, target_dataset)
+                
+                mesh_attrs = [
+                    "face_node_connectivity",
+                    "edge_node_connectivity",
+                    "edge_face_connectivity",
+                    "face_coordinates",
+                    "edge_coordinates",
+                    "node_coordinates",
+                ]
+                for mesh_attr in mesh_attrs:
+                    var_names = MapFile._get_var_names_from_var_attribute(mesh_variable, mesh_attr)
```

Comment:
maybe move to a method called: _copy_mesh

Reply:
Done